orte/mca/oob/tcp/oob_tcp_component.c

```diff
@@ -410,6 +410,37 @@ static int tcp_component_register(void)
                                           &mca_oob_tcp_component.disable_ipv6_family);
 #endif
 
+    
+    mca_oob_tcp_component.keepalive_time = -1;
+    (void)mca_base_component_var_register(component, "keepalive_time",
+                                          "Idle time in seconds before starting to send keepalives (num <= 0 => disable keepalive)",
+                                          MCA_BASE_VAR_TYPE_INT, NULL, 0, 0,
+                                          OPAL_INFO_LVL_9,
+                                          MCA_BASE_VAR_SCOPE_READONLY,
+                                          &mca_oob_tcp_component.keepalive_time);
+    if (0 < mca_oob_tcp_component.keepalive_time) {
+        struct protoent *proto;
+        if (NULL != (proto = getprotobyname("TCP"))) {
+            mca_oob_tcp_component.tcp_proto = proto->p_proto;
```

Comment:
why have `mca_oob_tcp_component.tcp_proto` and this call to `getprotobyname()` instead of just using the `IPPROTO_TCP` constant everywhere?  Is there some portability issue being dealt with here, or some sort of IPoIB compatibility?